inbm/dispatcher-agent/dispatcher/dispatcher_class.py

```diff
@@ -210,7 +210,7 @@ def _perform_startup_tasks(self) -> None:
                         create_snapshotter('update',
                                            snap_num='1',
                                            proceed_without_rollback=True,
-                                           ).commit()
+                                           reboot_device=True).commit()
```

Comment:
**Suggestion:** The proceed_without_rollback parameter is hardcoded to True which might not be the intended behavior, especially when the reboot_device parameter is also set to True. This could lead to a situation where a failed update cannot be rolled back, and the device is rebooted regardless. Consider using a configurable parameter or a condition check to determine the appropriate value for proceed_without_rollback. [maintainability]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
                        create_snapshotter('update',
                                           snap_num='1',
                                           proceed_without_rollback=self.proceed_without_rollback,
                                           reboot_device=self.should_reboot_device()).commit()
```

Reply:
It is intended.